fe/fe-core/src/main/java/com/starrocks/connector/jdbc/ClickhouseSchemaResolver.java

```diff
@@ -0,0 +1,379 @@
+     * │ hash_of_uncompressed_files            │ String          │              │                    │
+     * │ uncompressed_hash_of_compressed_files │ String          │              │                    │
+     * │ delete_ttl_info_min                   │ DateTime        │              │                    │
+     * │ delete_ttl_info_max                   │ DateTime        │              │                    │
+     * │ move_ttl_info.expression              │ Array(String)   │              │                    │
+     * │ move_ttl_info.min                     │ Array(DateTime) │              │                    │
+     * │ move_ttl_info.max                     │ Array(DateTime) │              │                    │
+     * │ default_compression_codec             │ String          │              │                    │
+     * │ recompression_ttl_info.expression     │ Array(String)   │              │                    │
+     * │ recompression_ttl_info.min            │ Array(DateTime) │              │                    │
+     * │ recompression_ttl_info.max            │ Array(DateTime) │              │                    │
+     * │ group_by_ttl_info.expression          │ Array(String)   │              │                    │
+     * │ group_by_ttl_info.min                 │ Array(DateTime) │              │                    │
+     * │ group_by_ttl_info.max                 │ Array(DateTime) │              │                    │
+     * │ rows_where_ttl_info.expression        │ Array(String)   │              │                    │
+     * │ rows_where_ttl_info.min               │ Array(DateTime) │              │                    │
+     * │ rows_where_ttl_info.max               │ Array(DateTime) │              │                    │
+     * │ projections                           │ Array(String)   │              │                    │
+     * │ bytes                                 │ UInt64          │ ALIAS        │ bytes_on_disk      │
+     * │ marks_size                            │ UInt64          │ ALIAS        │ marks_bytes        │
+     * └───────────────────────────────────────┴─────────────────┴──────────────┴────────────────────┴
+     *
+     * @param table
+     * @return
+     */
+    @NotNull
+    private String getPartitionQuery(Table table) {
+        String tableNameForPartInfo = getTableNameForPartInfo();
+        final String partitionQuery =
+                "SELECT  partition AS NAME, max(modification_time) AS MODIFIED_TIME FROM " + tableNameForPartInfo +
+                        " WHERE database = ? " + "AND table = ? AND name IS NOT NULL " +
+                        "GROUP BY partition ORDER BY partition";
+        if (table.isUnPartitioned()) {
+            throw new StarRocksConnectorException("GetPartitionQuery for unpartitioned table is not implemented");
+        } else {
+            return partitionQuery;
+        }
+    }
+
+    public static void main(String[] args) {
```

Comment:
testing code?

Reply:
Sorry about that, I am kind of using the git for synchronizing the code between local mac and my dev server, hence, the main for test. It's all cleaned up in the latest commit. With unittest added, and function working just fine in my environment.